server/src/main/java/org/elasticsearch/cluster/coordination/ElasticsearchNodeCommand.java

```diff
@@ -131,12 +129,11 @@ public static ClusterState clusterState(Environment environment, PersistedCluste
     protected void processNodePaths(Terminal terminal, OptionSet options, Environment env) throws IOException, UserException {
         terminal.println(Terminal.Verbosity.VERBOSE, "Obtaining lock for node");
         try (NodeEnvironment.NodeLock lock = new NodeEnvironment.NodeLock(logger, env, Files::exists)) {
-            final Path[] dataPaths =
-                    Arrays.stream(lock.getNodePaths()).filter(Objects::nonNull).map(p -> p.path).toArray(Path[]::new);
-            if (dataPaths.length == 0) {
+            if (lock.getNodePaths().length == 0) {
```

Comment:
`NodeEnvironment.NodeLock#nodePaths` always has length 1 now, so this will never happen. I'd prefer to refactor MDPs out of `NodeEnvironment` (including `NodeEnvironment.NodeLock`) first, because this was tricky to spot and there may be other things like it that I don't see.